packages/property-projector/src/components/ui/config-form/config-form.tsx

```diff
@@ -0,0 +1,393 @@
+import ColourPicker from '../colour-picker'
+import { ERROR_MESSAGES } from '@/constants/errors'
+import { SELLING_STATUS, LETTING_STATUS } from '@/constants/statuses'
+import { getNegotiatorOfficeId } from '@/util/negotiator-helper'
+import {
+  H5,
+  Section,
+  Formik,
+  Form,
+  FormSection,
+  FormHeading,
+  FormSubHeading,
+  Input,
+  Grid,
+  GridItem,
+  ImageInput,
+  Checkbox,
+  SelectBox,
+  SelectBoxOptions,
+  Button,
+  usePortal,
+  Alert,
+  DropdownSelect,
+  Loader,
+} from '@reapit/elements'
+
+type ConfigFormProps = {}
+
+const ConfigForm: React.FC<ConfigFormProps> = () => {
+  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
+
+  const [loading, setLoading] = useState(true)
+  const [config, setConfig]: any = useState(null)
+  const [allDepartments, setAllDepartments]: any[] = useState([])
+  const [allOffices, setAllOffices]: any[] = useState([])
+  const [officeId, setOfficeId] = useState('')
+
+  const [showProjector, hideProjector] = usePortal(() => <Projector config={config} />, [config])
+
+  console.info('Reapit Property Projector Config: ', config)
+
+  /**
+   * @todo adjust so the departments/offices aren't limited by pagination
+   */
+  useEffect(() => {
+    const fetchDepartments = async () => {
+      const departments = (await getDepartments(connectSession as ReapitConnectSession))?._embedded?.map(department => {
+        const { id, name, typeOptions: propertyTypes } = department
+
+        return {
+          id,
+          name,
+          propertyTypes,
+        }
+      })
+
+      setAllDepartments(departments)
+    }
+
+    const fetchOffices = async () => {
+      const offices = (await getOffices(connectSession as ReapitConnectSession))?._embedded?.map(department => {
+        const { id, name } = department
+
+        return {
+          id,
+          name,
+        }
+      })
+
+      setAllOffices(offices)
+    }
+
+    const fetchNegotiatorOfficeId = async () => {
+      setOfficeId(await getNegotiatorOfficeId(connectSession as ReapitConnectSession))
+    }
+
+    if (connectSession) {
+      console.log('Session Data:', connectSession)
+
+      Promise.all([fetchDepartments(), fetchOffices(), fetchNegotiatorOfficeId()]).catch(err =>
+        console.error(`Inital load error: ${err}`),
+      )
+
+      window.addEventListener('keydown', escapeKeyPressed, false)
+    }
+
+    return () => {
+      window.removeEventListener('keydown', escapeKeyPressed, false)
+    }
+  }, [connectSession])
+
+  useEffect(() => {
+    const fetchPropertyProjectorConfig = async () => {
+      setConfig(await getPropertyProjectorConfig(connectSession as ReapitConnectSession, officeId))
+    }
+
+    if (officeId !== '') {
+      fetchPropertyProjectorConfig().then(() => setLoading(false))
+    }
+  }, [officeId])
+
+  const escapeKeyPressed = event => {
+    if (event.keyCode !== 27) return
+    hideProjector()
+  }
+
+  const getInitialFormValues = () => {
```

Comment:
Consider breaking out some of the hook logic into separate files / functions for general readability and testing